### PR TITLE
58 alma update physical and electronic

### LINKED ISSUE #58: Update physical and electronic locations with alma new locations
see https://github.com/pulibrary/bibdata/issues/1368

### PR DESCRIPTION
**Pull request by christinach**
closes #58

### CODE REVIEW

**Comment by christinach** [comment]
This PR: 

- updates the physical holding location as 
`"holdings_1display": "{\"thesis\":{\"location\":\"Seeley G. Mudd Library (Mudd)\",\"library\":\"Seeley G. Mudd Library (Mudd)\",\"location_code\":\"mudd$stacks\",\"call_number\":\"AC102\",\"call_number_browse\":\"AC102\",\"dspace\":true}}",`

- and keeps the online theses records in the online facet by adding the field `electronic_portfolio_s` which is also used in bibdata to populate online links in the online section. 
`"electronic_portfolio_s": [
"{\"thesis\":{\"call_number\":\"AC102\",\"call_number_browse\":\"AC102\",\"dspace\":true}}"`]

**Comment by christinach** [comment]
A follow up issue is https://github.com/pulibrary/orangelight/issues/2486

**Comment by christinach** [comment]
examples: 
https://catalog-alma-qa.princeton.edu/catalog/dsp01jq085p03z/raw,
https://catalog-alma-qa.princeton.edu/catalog/dsp0141687h67f/raw,
https://catalog-alma-qa.princeton.edu/catalog/dsp0108612n756/raw

File: spec/visual_spec.rb [hunk 1]
```diff
@@ -81,7 +81,6 @@ def create_element(name, text)
       it 'excludes links that do not resolve' do
         links = JSON.parse(subject.send(:get_links, check_links))
         expect(links.has_key?('http://google.com')).to be true
-        expect(links.has_key?('http://libweb5.princeton.edu/visual_materials/ga/bad link.jpg')).to be false
```
Comment (hackartisan) [inline spec/visual_spec.rb:84]: This test doesn't seem related to the other stuff in the PR. I'm not sure why it's being deleted?

**Review by hackartisan — COMMENTED** [review]
See my one question in line.

Do we need to merge this before we can try indexing them into the alma qa site?

Do we not need a branch or anything here because we use release numbers?

File: spec/visual_spec.rb (hunk 1 above)
Comment (christinach) [reply spec/visual_spec.rb:84]: The test refers to a server that doesn't exist anymore. I confirmed with @kevinreiss .

Comment (christinach) [reply spec/visual_spec.rb:84]: > See my one question in line.
> 
> Do we need to merge this before we can try indexing them into the alma qa site?
> 
> Do we not need a branch or anything here because we use release numbers?

If we merge it we create a tag. The gem is referenced in the orangelight gemfile with a tag.
I indexed yesterday 3 example thesis records for @kevinreiss and @carolyncole to test them. They both confirmed in slack that they look ok with the requests and they can get to the aeon form.

Comment (hackartisan) [reply spec/visual_spec.rb:84]: great, thanks!

**Review by hackartisan — APPROVED** [review]
